src/hdmf/common/__init__.py

```diff
@@ -26,35 +27,34 @@
         is_method=False)
 def load_type_config(**kwargs):
     """
-    This method will either load the default config or the config provided by the path.
-    NOTE: This config is global and shared across all type maps.
+    This method will either load the config at the given path into either the global type map or a specific type map.
     """
     config_path = kwargs['config_path']
-    type_map = kwargs['type_map'] or get_type_map()
+    type_map = kwargs['type_map'] or __TYPE_MAP
```

Comment:
We want to use the global type map here, not a copy of the global type map. This worked before because copying the global type map maintained a reference to the same type configurator, so the configurator was unintentionally global.